cfme/common/vm_console.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+"""
+Module containing classes with common behaviour for consoles of both VMs and Instances of all types.
+"""
+
+import base64
+import re
+import tempfile
+
+from PIL import Image, ImageFilter
+from pytesseract import image_to_string
+from utils.log import logger
+from utils.pretty import Pretty
+from wait_for import wait_for, TimedOutError
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+
+banner_id = 'noVNC_status'  # specific for RHOS Provider
+canvas_id = 'noVNC_canvas'
+ctrl_alt_del_id = 'sendCtrlAltDelButton'  # specific for RHOS Provider
+connection_status_xpath = '//*[@id="connection-status"]'  # specific for the VMware/RHEV Provider
+canvas_xapth = '//*[@id="remote-console"]/canvas'  # specific to VMware/RHEV Provider
+ctrl_alt_del_xpath = '//*[@id="ctrlaltdel"]'  # specific to VMware/RHEV Provider
+
+
+class VMConsole(Pretty):
+    """Class to manage the VM Console.   Presently, only support HTML5 Console."""
+
+    pretty_attrs = ['appliance_handle', 'browser', 'console_handle', 'name']
+
+    def __init__(self, name, selenium, console_handle, appliance_handle, provider):
+        self.name = name
+        self.selenium = selenium
+        self.console_handle = console_handle
+        self.appliance_handle = appliance_handle
+        self.provider = provider
+
+    ###
+    # Methods
+    #
+    def _get_openstack_banner(self):
```

Comment:
**REQUIRED**:These methods here I would propose to move into the provider objects themselves. It can be called the same name then on ALL providers, needs no conditional, and is invoke by simply doing, ```self.provider.get_banner()``` If you chose to do it like that. You can also put a Base Provider object to raise NotImplemented exception. In this way, adding a new provider with a new way of getting the banner, requires no more changes to core files.